### DISCUSSION TITLE
关于模板相关的问题

### DISCUSSION THREAD

**Discussion by Zhaozhuofan** [BYJRK/DotNet-Discussions#243] [WPF 开发]
UP你好,我看了您原神那期的视频后想自己模仿着写一个QQ音乐
列表参考您原神的视频用的是RadioButton并重写了RadioButton的Tempalte
为Border里面套一个StackPanel之后再在Stack Panel里放了两个TextBlock
<img width="158" alt="image" src="https://github.com/BYJRK/DotNet-Discussions/assets/103978682/f1680dcc-b0a9-47e6-a7b9-69c3b89242d6">
<img width="900" alt="image" src="https://github.com/BYJRK/DotNet-Discussions/assets/103978682/21c3826a-e797-4482-a613-22640962571b">
<img width="571" alt="image" src="https://github.com/BYJRK/DotNet-Discussions/assets/103978682/b70ed467-ac2c-453b-870a-7f2409a3446f">
但是在使用trigger时提示我"无法识别或访问成员IsChecked,这是为什么呢?
跪求大佬解答

**Comment by Zhaozhuofan** [comment]
大佬你好,刚才这个问题解决了,写Property的时候写成RadioButton.IsChecked就好了.
但我还有个想请教的问题,是关于绑定的,向上面说的那样我把RadioButton的模板重写成了两个TextBlcok
我想把TextBlock的Text属性分别绑定为RadioButton的Content的一部分我该怎么做呢?
比如说RadioButton的Content是"1-2"的话,我想让两个TextBlock的Text分别为"1"和"2".
烦请大佬赐教.

**Comment by BYJRK** [comment]
你的第一个问题是因为你没有写 `ContentTemplate` 的 `TargetType`，所以它不知道 `IsChecked` 属性。你写成完整的依赖属性 `RadioButton.IsChecked`，或者它的基类 `ToggleButton.IsChecked` 都是可以的。第二个问题，如果你想在 XAML 中实现，或许可以将 `Content` 写成 `x:Array`。否则就需要写 Converter 之类的了。

**Reply by Zhaozhuofan** [reply]
谢谢您,学习了.
